src/clarity-angular/layout/vertical-nav/vertical-nav-group.ts

```diff
@@ -0,0 +1,186 @@
+                    } else if (!value && this.expandAnimationState === EXPANDED_STATE) {
+                        this.expandAnimationState = COLLAPSED_STATE;
+                    }
+                }
+            )
+        );
+
+        // If the nav is collapsing, close the open nav group + save its state
+        // If the nav is expanding, expand if the previous state was expanding
+        this._subscriptions.push(
+            this._verticalNavService.animateOnCollapsed.subscribe((goingToCollapse: boolean) => {
+                    if (goingToCollapse && this.expanded) {
+                        this.wasExpanded = true;
+                        this.expandAnimationState = COLLAPSED_STATE;
+                    } else if (!goingToCollapse && this.wasExpanded) {
+                        this.expandGroup();
+                        this.wasExpanded = false;
+                    }
+                }
+            )
+        );
+    }
+
+    private wasExpanded: boolean = false;
+
+    @Input("clrVerticalNavGroupExpanded")
+    set userExpandedInput(value: boolean) {
+        value = !!value;
+        if (this.expanded !== value) {
+            // We have to call toggleExpand because some cases require animations to occur first
+            // Directly setting the Expand service value skips the animation and can result in
+            // nodes in the DOM but the nav group still being collapsed
+            this.toggleExpand();
+        }
+    }
+
+    get expanded(): boolean {
+        return this._itemExpand.expanded;
+    }
+
+    set expanded(value: boolean) {
+        if (this._itemExpand.expanded !== value) {
+            this._itemExpand.expanded = value;
+            this.expandedChange.emit(value);
+        }
+    }
+
+    @Output("clrVerticalNavGroupExpandedChange")
+    expandedChange: EventEmitter<boolean> = new EventEmitter<boolean>(true);
+
+    @Input("clrVerticalNavGroupActive")
+    set active(value: boolean) {
+        value = !!value;
+        if (this._active !== value) {
+            if (value) {
+                this._verticalNavService.deactivateOtherNavGroups(this);
+            }
+            this._active = value;
+        }
+    }
+
+    private _subscriptions: Subscription[] = [];
+
+    private _expandAnimationState: string = COLLAPSED_STATE;
+
+    expandGroup(): void {
+        this.expanded = true;
+        // Expanded animation occurs after Expand.expand is set to true
+        this.expandAnimationState = EXPANDED_STATE;
+    }
+
+    collapseGroup(): void {
+        // If a Vertical Nav Group toggle button is clicked while the Vertical Nav is in Collapsed state,
+        // the Vertical Nav should be expanded first.
+        this.expandAnimationState = COLLAPSED_STATE;
+    }
+
+    // closes a group after the collapse animation
+    expandAnimationDone($event: any) {
+        if ($event.toState === COLLAPSED_STATE) {
+            this.expanded = false;
+        }
```

Comment:
This is a bit of a mess, you have 2 sources of truth for the expanded status: this component, and the `Expand` service. And they don't even agree because of animations.

As I mentioned last time, we need to update the `IfExpanded` feature so that it handles animations correctly, because we have the problem on all of these components. I don't understand why this would have issues that TreeView doesn't have, since it really is just a tree. So either we missed issues in TreeView, or we're doing something overkill here.